### DISCUSSION TITLE
Add proper single page web-routing into NiceGUI?

### DISCUSSION THREAD

**Discussion by AlexKotik** [zauberzeug/nicegui#1912] [Ideas / Feature Requests]
It would be great to have a proper single page routing (without page refresh) built into the library itself (without the need for custom JS code). I've seen an example for single page routing, but it relies on custom router class and JS-code, it would be so much nicer if NiceGUI would abstract that away using decorators or something. Thanks.

**Comment by falkoschindler** [comment]
Yes, that would definitely be great!

As far as I remember, on reason why the Router landed in the [Single Page App example](https://github.com/zauberzeug/nicegui/tree/main/examples/single_page_app) and didn't become official API was its name. Even though "router" is pretty common in frontend frameworks, it is already used in FastAPI apps, so we would need to find something different.

Apart from that I think the `Router` class from the example is already pretty good and could be released. What do you think, @rodja?

**Comment by rodja** [comment]
This is a duplicate of https://github.com/zauberzeug/nicegui/discussions/1066, let's continue there.
